modules/protocol_gateways/setup_s3.sh

```diff
@@ -0,0 +1,49 @@
+echo "$(date -u): running S3 script"
+
+# get all protocol gateways frontend container ids separated by comma
+all_container_ids_str=$(echo "$all_container_ids" | tr '\n' ',' | sed 's/,$//')
+
+function retry_create_s3_cluster {
+  retry_max=60
+  retry_sleep=30
+  count=$retry_max
+  check_s3_cluster_status
+  while [ $count -gt 0 ]; do
+      weka s3 cluster create $filesystem_name .config_fs --container $all_container_ids_str --port 9000 && break
+      count=$(($count - 1))
+      echo "Retrying create S3 cluster in $retry_sleep seconds..."
+      sleep $retry_sleep
+      check_s3_cluster_status && break
+  done
+  [ $count -eq 0 ] && {
+      echo "create S3 cluster command failed after $retry_max attempts"
+      echo "$(date -u): create S3 cluster failed"
+      return 1
+  }
+  return 0
+}
+
+if [[ `weka s3 cluster status` ]]; then
```

Comment:
you should avoid this syntax
see: https://github.com/koalaman/shellcheck/wiki/SC2092
and https://github.com/koalaman/shellcheck/wiki/SC2091